### ISSUE TITLE
"Returned address does not exist"

### ISSUE THREAD

**Issue by the-real-aquaman** [eris-ltd/eris-contracts.js#49]
[eris-db: 0.15.5]
[eris-contracts: 0.15.5]
[node.js: 6.9.1]

After creating a new contract from within another contract and returning the address of that, I get an error saying the address does not exist. Here's my code: 
```
contract Person {

    address _owner;
    string _firstName;
    string _lastName;
    uint _age;
    Pet[] _pets;

    function Person(string firstName, string lastName, uint age) {
        _firstName = firstName;
        _lastName = lastName;
        _age = age;
        _owner = msg.sender;
    } 
...

contract PersonAdministration {

    Person[] _persons;

    function createPerson(string firstName, string lastName, uint age) constant returns (address) {
        Person person = new Person(firstName, lastName, age);
        _persons.push(person);
        return address(person);
    } 
...
```

JavaScript:

```
personadminContract.createPerson("John", "Smith", 44, function(error, personAddress) {
                if (error) {
                    throw error;
                }   
                console.log(personAddress);
                contractsFactoryPerson.at(personAddress, function(error, returnedPerson){
                    if (error) {
                        throw error;
                    }
                    returnedPerson.getFirstName(function(error, success) {
                        console.log('error', error)
                        if (error) {
                            throw error;
                        }
                        console.log(success);
```

The error occurs right after the console.log(personAddress):
```
2E1AB875DA8E624F019E8295D586615F9FB4888D
Call of Eris DB method 'call' with parameters {
  "from": "4E932AF562C9924F34941AC3491970D52A854013",
  "address": "2E1AB875DA8E624F019E8295D586615F9FB4888D",
  "data": "932997f1"
} responded with "Account 2E1AB875DA8E624F019E8295D586615F9FB4888D does not exist".
error { Error: Account 2E1AB875DA8E624F019E8295D586615F9FB4888D does not exist
    at values.next.then (/home/pieter/workspace/BC_Test/node_modules/eris-contracts/node_modules/eris-db/node_modules/@nodeguy/json-rpc/lib/index.js:12:38)
  method: 'call',
  params: 
   [ { from: '4E932AF562C9924F34941AC3491970D52A854013',
       address: '2E1AB875DA8E624F019E8295D586615F9FB4888D',
       data: '932997f1' } ],
  code: -32603,
  message: 'Account 2E1AB875DA8E624F019E8295D586615F9FB4888D does not exist' }
(node:6506) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Account 2E1AB875DA8E624F019E8295D586615F9FB4888D does not exist

```

If you look at the last line, it says UnhandledPromiseRejection. Also when installing the node modules in node.js v6,9.1, there is an unmet dependency for request@^2.34 which is a peer for request-promise-core. Could this be the issue? Also read in the slack forums about this [#apps-js]:

> nya	[7:36 PM]  
> @nodeguy ok now it finally works on 0.15.4 but only for node version < 6.6.0 because of the deprecated promise reject that event is using atm

**Comment by somak12** [comment]
I am also facing the same issue. Even if the address of the deployed contract exist in the epm.json, when calling the contract at the specific address it says "Account does not exist"

**Comment by somak12** [comment]
which node version is it working for?

**Comment by NodeGuy** [comment]
Please try this again with Eris DB 0.12.1.  You can tell the CLI to use the newer version by editing your `~/.eris/eris.toml` file and looking for the following line:

```
ImageDB = "eris/db:0.12.0"
```

Change it to read:

```
ImageDB = "eris/db:0.12.1"
```

Create a new chain and then try it again.

**Comment by somak12** [comment]
I have tried in new eris 0.12.0, using eris-contract.js 0.15.6 during npm it gives error  'UNMET PEER DEPENDENCY request@^2.34' 
When I try to do node app.js it gives error 

/usr/src/app/node_modules/eris-contracts/lib/contractManager.js:108
  const wrapMethod = (name, outputUnpacker = R.identity) =>
                                           ^

SyntaxError: Unexpected token =
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/usr/src/app/node_modules/eris-contracts/index.js:1:85)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)

Node : 6.9.1
npm: 3.10.8

Am I doing something wrong here?

**Comment by NodeGuy** [comment]
Please follow my directions above.

**Comment by somak12** [comment]
Just for clarification you mean to say  'UNMET PEER DEPENDENCY request@^2.34' will get resolved when I use the ImageDB = "eris/db:0.12.1"?? The above error "Account doesnot exist" was occurring in previous version of eris.
